viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/process/strategy/TerminatedTwoTrackSectionSynchronizer.scala

```diff
@@ -0,0 +1,473 @@
+   */
+  def roadPartStartTerminated(projectLinks: Seq[ProjectLink]): Seq[ProjectLink] = {
+    val terminatedLinks = projectLinks.filter(_.status == RoadAddressChangeType.Termination)
+
+    val (leftTerminatedSections, rightTerminatedSections) = terminatedLinksToToContinuousTwoTrackSections(terminatedLinks)
+
+    val roadPartStartTerminatedLefts = leftTerminatedSections.find(section => section.exists(_.addrMRange.isRoadPartStart))
+    val roadPartStartTerminatedRights = rightTerminatedSections.find(section => section.exists(_.addrMRange.isRoadPartStart))
+
+    val processedLinks = if (roadPartStartTerminatedLefts.nonEmpty && roadPartStartTerminatedRights.nonEmpty) { // Check if road part start is terminated on both tracks
+      handleTwoTrackRoadPartStartTermination(roadPartStartTerminatedLefts.get, roadPartStartTerminatedRights.get, projectLinks)
+    } else
+      projectLinks
+
+    processedLinks
+  }
+
+  /**
+   * Adjusts terminated sections on two track road part start to match (if reasonably close to each other) and
+   *  the links that come after the terminated section.
+   *
+   * Example:
+   *
+   *  Terminated:     ==>
+   *  Nonterminated:  -->
+   *
+   *  Before:
+   *
+   *  0       202  250   450
+   *  ========>---->----->
+   *  ======>------->---->
+   *  0     200     260  450
+   *
+   *  After:
+   *
+   *  0       201  250   450
+   *  ========>---->----->
+   *  ======>------->---->
+   *  0     201     260  450
+   */
+  def handleTwoTrackRoadPartStartTermination(terminatedLeftSection: Seq[ProjectLink], terminatedRightSection: Seq[ProjectLink], projectLinks: Seq[ProjectLink]): Seq[ProjectLink]= {
+    val maxDiffForTrackEnds = 10 // This number is arbitrary and may require adjustments in the future.
+    val lastTerminatedOnLeftSideSection = terminatedLeftSection.last
+    val lastTerminatedOnRightSideSection = terminatedRightSection.last
+
+    val continuousAfterTerminatedLeft   = projectLinks.find(pl => pl.track == Track.LeftSide && pl.originalAddrMRange.continuesFrom(lastTerminatedOnLeftSideSection.originalAddrMRange))
+    val continuousAfterTerminatedRight  = projectLinks.find(pl => pl.track == Track.RightSide && pl.originalAddrMRange.continuesFrom(lastTerminatedOnRightSideSection.originalAddrMRange))
+
+    // Adjust terminated
+    val processedLinks: Seq[ProjectLink] = {
+      if (Math.abs(lastTerminatedOnLeftSideSection.addrMRange.end - lastTerminatedOnRightSideSection.addrMRange.end) <= maxDiffForTrackEnds) {
+        // Adjust the last terminated links on each track to match
+        val (adjustedTerminatedLeft, adjustedTerminatedRight, averagedAddrMRange) = adjustTerminatedToMatch(lastTerminatedOnLeftSideSection, lastTerminatedOnRightSideSection)
+
+        if (continuousAfterTerminatedLeft.isDefined && continuousAfterTerminatedRight.isDefined) {
+          // Adjust both links and update the project links list with the adjusted terminated links and the adjusted continuous-after-termination links
```

Comment:
// Adjust both links after the terminated section

and put the rest of the comment below, right before the update